.github/workflows/ci.yml

```diff
@@ -13,80 +13,30 @@ jobs:
+          - {name: "Ubuntu GCC 14", tag: "gnu-14", os: ubuntu-24.04, toolchain: "gcc-14", cmake_args: "-G \"Ninja Multi-Config\" -DCMAKE_CONFIGURATION_TYPES=\"RelWithDebInfo;Asan;Gcov\" ", coverage: true}
+          - {name: "Ubuntu GCC 13", tag: "gnu-13", os: ubuntu-24.04, toolchain: "gcc-13",  cmake_args: "-G \"Ninja Multi-Config\" -DCMAKE_CONFIGURATION_TYPES=\"RelWithDebInfo;Asan\" "}
+          - {name: "Ubuntu GCC 12", tag: "gnu-12", os: ubuntu-24.04, toolchain: "gcc-12",  cmake_args: "-G \"Ninja Multi-Config\" -DCMAKE_CONFIGURATION_TYPES=\"RelWithDebInfo;Asan\" "}
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: 'true'
-      - uses: seanmiddleditch/gha-setup-ninja@master
-      - name: Activate verbose shell
-        run: set -x
-      - name: Install LLVM+Clang
-        if: startsWith(matrix.config.name, 'Ubuntu Clang')
-        run: |
-          set -x
-          cat /etc/lsb-release
-          # Remove existing Clang installations.
-          sudo apt-get remove \
-            clang-${{matrix.config.installed_clang_version}} \
-            clang++-${{matrix.config.installed_clang_version}} \
-            clangd-${{matrix.config.installed_clang_version}} \
-            clang-tidy-${{matrix.config.installed_clang_version}} \
-            clang-format-${{matrix.config.installed_clang_version}} \
-            clang-tools-${{matrix.config.installed_clang_version}} \
-            llvm-${{matrix.config.installed_clang_version}}-dev \
-            lld-${{matrix.config.installed_clang_version}} \
-            lldb-${{matrix.config.installed_clang_version}} \
-            llvm-${{matrix.config.installed_clang_version}}-tools \
-            libc++-${{matrix.config.installed_clang_version}}-dev \
-            libc++abi-${{matrix.config.installed_clang_version}}-dev \
-            libclang-common-${{matrix.config.installed_clang_version}}-dev \
-            libclang-${{matrix.config.installed_clang_version}}-dev \
-            libclang-cpp${{matrix.config.installed_clang_version}}-dev \
-            libomp-${{matrix.config.installed_clang_version}}-dev \
-            libunwind-${{matrix.config.installed_clang_version}}-dev \
-            libc++-dev libc++1 libc++abi-dev libc++abi1
-          # Install LLVM+Clang.
-          CLANG_VERSION=$(echo ${{matrix.config.toolchain}} | cut -d '-' -f2)
-          wget https://apt.llvm.org/llvm.sh
-          chmod +x llvm.sh
-          sudo ./llvm.sh ${CLANG_VERSION} all
-          # Link Clang libraries (if not done by llvm.sh - some links are already set).
-          sudo ln -fs /usr/lib/llvm-${CLANG_VERSION}/lib/lib* /usr/lib/x86_64-linux-gnu/ || true
-          # If Clang 17, install a newer version of libc++ and libc++abi.
-          [[ ${CLANG_VERSION} = 17 ]] && sudo apt-get install libc++-dev libc++1 libc++abi-dev libc++abi1
-          find /usr/lib/x86_64-linux-gnu/ -name libc++.so* || true
-          clang++-${CLANG_VERSION} --version
```

Comment:
What is the rational for special treatment for clang 17?

Are we only supporting clang 17 with an updated libc++ install?

@steve-downey @neatudarius